### PR TITLE
RWA-576: Migrate to new role assignment v2 query endpoint, update tests and pact

### PR DESCRIPTION
**Pull request by alexbaeza**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-576

### Change description ###

RWA-576: Migrate to new role assignment v2 query endpoint, update tests and pact

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: charts/wa-task-configuration-api/values.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ java:
   aadIdentityName: wa
   environment:
     CAMUNDA_URL: "http://camunda-api-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal/engine-rest"
-    ROLE_ASSIGNMENT_URL: "http://am-role-assignment-service-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
+    ROLE_ASSIGNMENT_URL: "http://am-role-assignment-service-pr-833.service.core-compute-preview.internal"
```
Comment (alexbaeza) [inline charts/wa-task-configuration-api/values.yaml:8]:
```suggestion
    ROLE_ASSIGNMENT_URL: "http://am-role-assignment-service-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
```

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -41,7 +41,7 @@ withPipeline(type, product, component) {
   env.S2S_URL = "http://rpe-service-auth-provider-aat.service.core-compute-aat.internal"
   env.CAMUNDA_URL = "http://camunda-api-aat.service.core-compute-aat.internal/engine-rest"
   env.CCD_URL = "http://ccd-data-store-api-aat.service.core-compute-aat.internal"
-  env.ROLE_ASSIGNMENT_URL = "http://am-role-assignment-service-aat.service.core-compute-aat.internal"
+  env.ROLE_ASSIGNMENT_URL = "http://am-role-assignment-service-pr-833.service.core-compute-preview.internal"
```
[the remark this answers is not in the document]
Comment (alexbaeza) [reply Jenkinsfile_CNP:44]: The suggestion above needs to be committed before merging to master.

File: charts/wa-task-configuration-api/values.yaml (hunk 1 above)
Comment (alexbaeza) [reply charts/wa-task-configuration-api/values.yaml:8]: The suggestion above needs to be committed before merging to master.

**Review by indvamshi — APPROVED** [review]

File: Jenkinsfile_CNP (hunk 2 above)
Comment (iguannago) [inline Jenkinsfile_CNP:44]: we need to remember to revert this change

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -84,12 +84,15 @@ launchDarkly:
 logging:
   level:
     root: INFO
+    uk.gov.hmcts.reform.wataskmanagementapi.clients.*: DEBUG
```
Comment (iguannago) [inline src/main/resources/application.yaml:87]: are we sure about turning ON debug mode in Prod?

Comment (alexbaeza) [reply src/main/resources/application.yaml:87]: This is only for the feign clients same thing is enabled on task management, it allows us to see each call being made and it's response

**Review by iguannago — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/wataskconfigurationapi/auth/role/entities/request/QueryRequest.java [hunk 4]
```diff
@@ -32,6 +33,8 @@
     private final List<RoleCategory> roleCategory;
     private final Map<String, List<String>> attributes;
     private final List<String> authorisations;
+    private final List<String> hasAttributes;
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskconfigurationapi/auth/role/entities/request/QueryRequest.java:36]: could this be named better sounds like a boolean?

File: src/main/java/uk/gov/hmcts/reform/wataskconfigurationapi/auth/role/entities/request/QueryRequest.java [hunk 5]
```diff
@@ -32,6 +33,8 @@
...
+    private final Boolean readOnly;
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskconfigurationapi/auth/role/entities/request/QueryRequest.java:37]: Maybe consider a native boolean?  Could this be null or should we default to false?

File: src/main/java/uk/gov/hmcts/reform/wataskconfigurationapi/auth/role/RoleAssignmentService.java [hunk 6]
```diff
@@ -63,13 +64,16 @@ public RoleAssignmentResource performSearch(QueryRequest queryRequest) {
         }
     }
 
-    private QueryRequest buildQueryRequest(String caseId) {
-        return QueryRequest.builder()
+    private MultipleQueryRequest buildQueryRequest(String caseId) {
+        QueryRequest queryRequest = QueryRequest.builder()
             .roleType(singletonList(RoleType.CASE))
             .roleName(singletonList("tribunal-caseworker"))
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskconfigurationapi/auth/role/RoleAssignmentService.java:70]: Not part of this PR but are we hard coding the roleName?  What if we want to query for a different role(s).

**Review by iguannago — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/wataskconfigurationapi/auth/role/RoleAssignmentService.java (hunk 6 above)
Comment (alexbaeza) [reply src/main/java/uk/gov/hmcts/reform/wataskconfigurationapi/auth/role/RoleAssignmentService.java:70]: I believe this was part of a previous ticket to find if someone should be auto-assigned if we wanted to support more roles might need to review if it is a change of requirements.

File: src/main/java/uk/gov/hmcts/reform/wataskconfigurationapi/auth/role/entities/request/QueryRequest.java (hunk 5 above)
Comment (alexbaeza) [reply src/main/java/uk/gov/hmcts/reform/wataskconfigurationapi/auth/role/entities/request/QueryRequest.java:37]: These are the Am-role assignment entities, however, we could rename them within our app
